### DISCUSSION TITLE
Does github actions run on non mergable pull request

### DISCUSSION THREAD

**Discussion by david-gang** [community/community#25904] [Actions]
<p>I have an action which runs on pull_request trigger. What happens with pull requests which are not mergable? Will the action still be triggered?</p>

**Comment by NicoNBW** [comment]
<p>Hi <a class="mention" href="/david-gang">@david-gang</a><br>
Thanks for your feedback.<br>
Yes, the action will still be triggered, because more than one activity type triggers this event. such as:</p>
<pre><code class="lang-auto">on:
  pull_request:
    types: [assigned, opened, synchronize, reopened]
</code></pre>
<p>More about pull_request activity types please refer <a href="https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request" rel="nofollow noopener">here</a>.</p>

**Comment by david-gang** [comment]
<p>Can you please tell me which activity type is active in this case? Thanks</p>

**Comment by NicoNBW** [comment, marked as answer]
<p>Hi <a class="mention" href="/david-gang">@david-gang</a><br>
I misunderstood you in the reply above. Sorry for that. I update the reply:<br>
If there is a conflict, your action will not be triggered. It will be triggered until the conflict is resolved.<br>
</p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/d/d9815bd4f4ace38ffd6209489b8432bdb4e21580.png" data-download-href="/uploads/short-url/v28LaKX3dCT3CuWVatsGXPDuQfK.png?dl=1" title="image" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/46212442/181094055-e4a32cdb-7272-4f2f-aa55-0d921e527ef8.png" alt="image" data-base62-sha1="v28LaKX3dCT3CuWVatsGXPDuQfK" width="690" height="175" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/d/d9815bd4f4ace38ffd6209489b8432bdb4e21580_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">image</span><span class="informations">919×234 12.3 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div><br>
If you set the Branch protection rule and some <a href="https://docs.github.com/en/github/administering-a-repository/about-required-status-checks" rel="nofollow noopener">Require status checks</a> fail, the pull request is also in a non-mergable state. In this case, the action still be triggered as long as <a href="https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull_request" rel="nofollow noopener">the pull request activity types</a> are met.
